content/groth16/en/groth16.md

```diff
@@ -299,10 +299,10 @@ The equation above is valid, but the witness does not necessarily satisfy the or
 
 Therefore, we need to prevent the prover from using $\Psi_1$ to $\Psi_{\ell}$ as part of the computation of $[C]_1$.
 
-### Introducing $\gamma$ and $\delta$
-To avoid the problem above, the trusted setup introduces new scalars $\gamma$ and $\delta$ to force $\Psi_{\ell+1}$ to $\Psi_m$ to be separate from $\Psi_1$ to $\Psi_{\ell}$. To do this, the trusted setup divides (multiplies by the modular inverse) the private terms (that constitute $[C]_1$) by $\delta$ and the public terms (that constitute $[X]_1$, the sum the verifier computes) by $\gamma$.
+### Introducing $\gamma$ and/or $\delta$
+To avoid the problem above, the trusted setup introduces a new scalar : $\gamma$ or $\delta$ to force $\Psi_{\ell+1}$ to $\Psi_m$ to be separate from $\Psi_1$ to $\Psi_{\ell}$. To do this, the trusted setup divides (multiplies by the modular inverse) the private terms (that constitute $[C]_1$) by $\delta$ and/or the public terms (that constitute $[X]_1$, the sum the verifier computes) by $\gamma$.
 
-Since the $h(\tau)t(\tau)$ term is embedded in $[C]_1$, those terms also need to be divided by $\delta$.
+Since the $h(\tau)t(\tau)$ term is embedded in $[C]_1$, those terms also need to be divided by $\delta$. If either $\delta$ and $\gamma$ have an unknown discrete logarithm, then the forgery described earlier along possible other methods are avoided. This method was used in Zcash’s Sapling’s‑based [trusted setups](https://github.com/ebfull/phase2/blob/master/src/lib.rs#L808) where $\gamma$ is simply left to $G_2$ and $\delta$ is still randomly generated from $G_2$.
```

Comment:
the phrase "randomly generated from $G_2$" doesn't quite describe a concrete operation.

I believe you might have meant "randomly generate from $G_2$" means "we multiplied $G_2$ by a random number" but it isn't quite saying that.

Some alternatives:
- is randomly chosen during the trusted setup, then published as $[\gamma G_2]$
- is some other random point in the $\mathbb{G}_2$ elliptic curve group

Reply:
@RareSkills No. What I’m meaning is values are generated according to G₂ then such linked values are updated from G₂ when G₂ become randomized into γ or δ. For example, the points used in G₁ for public inputs are generated first according to G₂ and then when γ is randomized, the public inputs points are updated.

Do you have a better way to say this ?

Reply:
I think where I'm confused is "generated according to G₂"

What does it mean to generate according to a certain point (the generator int his case)?

Reply:
@RareSkills you generate values first in G₁, then when you apply randomness on δ or γ,  you apply the same scalar on the previously generated G₁ values.